internal/controllers/machinedeployment/machinedeployment_controller_test.go

```diff
@@ -276,6 +307,57 @@ func TestMachineDeploymentReconciler(t *testing.T) {
 			return len(machineSets.Items)
 		}, timeout).Should(BeEquivalentTo(2))
 
+		// Update the Labels of the MachineDeployment, expect Reconcile to be called and the MachineSet to be updated in-place.
+		t.Log("Setting a label on the MachineDeployment")
+		modifyFunc = func(d *clusterv1.MachineDeployment) { d.Spec.Template.Labels["updated"] = "true" }
+		g.Expect(updateMachineDeployment(ctx, env, deployment, modifyFunc)).To(Succeed())
+		g.Eventually(func() map[string]string {
+			if err := env.List(ctx, machineSets, msListOpts...); err != nil {
+				return nil
+			}
+			return machineSets.Items[0].Spec.Template.Labels
+		}, timeout).Should(HaveKeyWithValue("updated", "true"))
+
+		// Verify that the old machine set does not get the updated labels.
+		g.Expect(machineSets.Items[1].Spec.Template.Labels).ShouldNot(HaveKeyWithValue("updated", "true"))
```

Comment:
```suggestion
		g.Eventually(func(g Gomega) {
			g.Expect(env.List(ctx, machineSets, msListOpts...)).To(Succeed())

			// Verify we still only have 2 MachineSets.
			g.Expect(machineSets.Items).To(HaveLen(2))
			// Verify that the new MachineSet gets the updated labels.
			g.Expect(machineSets.Items[0].Spec.Template.Labels).To(HaveKeyWithValue("updated", "true"))
			// Verify that the old MachineSet does not get the updated labels.
			g.Expect(machineSets.Items[1].Spec.Template.Labels).ShouldNot(HaveKeyWithValue("updated", "true"))
		}, timeout).Should(Succeed())
```
WDYT about this instead? Should be more robust and also verifies that we still only have 2 MachineSets

Reply:
Can we do the same with only one Eventually below for the timeouts?